### ISSUE TITLE
Apollo Unable to generate classes for operations with Default Arguments on v3.2.0

### ISSUE THREAD

**Issue by AnkitPatanaik** [apollographql/apollo-kotlin#4004]
**Summary**
Hi again! I'm running into an issue where Apollo is unable to generate classes when there are default arguments in my Query/Mutation. It specifically throws this error: 

```
Caused by: java.lang.IllegalArgumentException: couldn't make a guess for [com.apollographql.relocated.cl](http://com.apollographql.relocated.cl/)
```

**Version**
3.2.0, things were working as expected on 3.1.0

**Description**

**These queries don't work**

_Mandatory Arguments, no defaults_
```
mutation MyMutation($firstName: CustomScalar!, $lastName: CustomScalar!) {
  my_put_operation(firstName: $firstName, lastName: $lastName)
}
```
_Hardcoded default at field level w/ mandatory arguments_
```
mutation MyMutation($firstName: CustomScalar!, $lastName: CustomScalar!) {
  my_put_operation(firstName: "MyDefault123" lastName: $lastName)
}
```

**The following don't work**

_Default for non-mandatory argument for mutation_
```
mutation MyMutation($firstName: CustomScalar = "MyDefault123", $lastName: CustomScalar!) {
  my_put_operation(firstName: $firstName, lastName: $lastName)
}
```
_Non-Mandatory argument, default value hardcoded in field_
```
mutation MyMutation($firstName: CustomScalar , $lastName: CustomScalar!) {
  my_put_operation(firstName: "MyDefault123" lastName: $lastName)
}
```

**Here's the full stack trace for more context:**
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':subsystem:myFeature:generateServiceApolloSources'.
> couldn't make a guess for com.apollographql.relocated.cl

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 34s
53 actionable tasks: 3 executed, 50 up-to-date

Publishing build scan...

ERRORS (1)

FAILED TASK: :subsystem:myFeature:generateServiceApolloSources
  Description: org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':subsystem:myFeature'.

    Caused by: java.lang.IllegalArgumentException: couldn't make a guess for com.apollographql.relocated.cl
      // ...
      at com.apollographql.relocated.d4.a(SourceFile:7)
      at com.apollographql.relocated.g4.a(SourceFile:217)
      at com.apollographql.apollo3.compiler.codegen.kotlin.helpers.DataClassKt.makeDataClass(SourceFile:13)
      at com.apollographql.apollo3.compiler.codegen.kotlin.file.OperationBuilder.typeSpec(SourceFile:844)
      at com.apollographql.apollo3.compiler.codegen.kotlin.file.OperationBuilder.build(SourceFile:3)
      at com.apollographql.apollo3.compiler.codegen.kotlin.KotlinCodeGen.write(SourceFile:206)
      at com.apollographql.apollo3.compiler.ApolloCompiler.write(SourceFile:790)
      at com.apollographql.apollo3.gradle.internal.ApolloGenerateSourcesTask.taskAction(SourceFile:275)
      // ...
```

Let me know if there's anything else I could add that would be helpful!

**Comment by BoD** [comment]
Hi! 👋  Thanks for reporting this.

For now, I couldn't reproduce the issue and the message "couldn't make a guess" doesn't ring a bell.

I have created a very simple project [here](https://github.com/BoD/apollo-kotlin-repro-issue-4004), maybe you can try and tweak it to show how to reproduce the issue? Thanks a lot!

**Comment by BoD** [comment]
Actually we've found the culprit (thanks @martinbonnin!), it is due to using `addJvmOverloads` which unfortunately doesn't work as expected due to relocation.

This will need a fix on our part, in the meantime you can disable `addJvmOverloads` and this should resolve this issue.

**Comment by theBradfo** [comment]
Ahh good find.  I wonder why the tests included as part of #3907 didn't fail originally 🤔

**Comment by BoD** [comment]
@theBradfo This is because these don't depend on the relocated version of the plugin/compiler.

**Comment by AnkitPatanaik** [comment]
Just confirming that removing `addJvmOverloads` does remedy the issue, however I would like to use `addJvmOverloads` eventually so a fix would be much appreciated :)

**Comment by martinbonnin** [comment]
Fixed in just released 3.2.2
